extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/AbstractJpaOperations.java

```diff
@@ -421,7 +431,43 @@ public boolean exists(Class<?> entityClass, String query, Parameters params) {
     }
 
     public long deleteAll(Class<?> entityClass) {
-        return (long) getEntityManager(entityClass).createQuery("DELETE FROM " + getEntityName(entityClass)).executeUpdate();
+        //detecting the case where there are cascade-delete associations, and do the the bulk delete query otherwise.
+        if (deleteOnCascadeDetected(entityClass)) {
+            int count = 0;
+            List<?> objects = list(findAll(entityClass));
+            for (Object entity : objects) {
+                getEntityManager().remove(entity);
+                count++;
+            }
+            return count;
+        }
+        return (long) getEntityManager().createQuery("DELETE FROM " + getEntityName(entityClass)).executeUpdate();
+
+    }
+
+    /**
+     * Detects if cascading delete is needed. The delete-cascading is needed when associations with cascade delete enabled
+     * {@link javax.persistence.OneToMany#cascade()} and also on entities containing a collection of elements
+     * {@link javax.persistence.ElementCollection}
+     * 
+     * @param entityClass
+     * @return true if cascading delete is needed. False otherwise
+     */
+    private boolean deleteOnCascadeDetected(Class<?> entityClass) {
+        Metamodel metamodel = getEntityManager().getMetamodel();
+        EntityType<?> entity1 = metamodel.entity(entityClass);
+        Set<Attribute<?, ?>> declaredAttributes = ((EntityTypeImpl) entity1).getDeclaredAttributes();
+
+        CascadeStyle[] propertyCascadeStyles = getEntityManager().unwrap(SessionImplementor.class)
+                .getEntityPersister(entityClass.getName(), null)
+                .getPropertyCascadeStyles();
+        boolean doCascade = Arrays.stream(propertyCascadeStyles)
+                .filter(cascadeStyle -> cascadeStyle.doCascade(CascadingActions.DELETE))
+                .count() > 0;
```

Comment:
Very minor, but I think you can use `anyMatch()` here, no?

Reply:
yes, of course. Indeed, I used firstly the anyMatch but change because I was using another method and forget to use it again after refactor.